rclpy/src/rclpy/node.cpp

```diff
@@ -478,6 +492,21 @@ Node::Node(
   if (RCL_RET_OK != ret) {
     throw RCLError("error creating node");
   }
+
+  if (rcl_logging_rosout_enabled() && enable_rosout) {
+    rclpy::LoggingGuard scoped_logging_guard;
+    ret = rcl_logging_rosout_init_publisher_for_node(rcl_node_.get());
+    if (ret != RCL_RET_OK) {
+      if (rcl_logging_rosout_fini_publisher_for_node(rcl_node_.get()) != RCL_RET_OK) {
```

Comment:
If the return value of `rcl_logging_rosout_init_publisher_for_node` is not RCL_RET_OK, I don't think we need to call `rcl_logging_rosout_fini_publisher_for_node`.